### ISSUE TITLE
GitHub branch configuration

### ISSUE THREAD

**Issue by GitMensch** [OCamlPro/gnucobol#51]
I suggest to do the following:
* [change default branch](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch) to gcos4gnucobol-3.x
* [add branch protection](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule#creating-a-branch-protection-rule) to master and gnucobol-3.x

**Comment by nberth** [comment]
I agree the purpose of this repository is mainly about branch `gcos4gnucobol-3.x`.
@lefessan do you think protecting branches `master` and `gnucobol-3.x` would break your mechanics for importing upstream branches?

**Comment by lefessan** [comment]
Unfortunately, I don't think I can protect the branches as the sync process pushes directly into these branches.

I am going to push a base branch (maybe with another name than `gcos4gnucobol`) with the git-related commits.

**Comment by GitMensch** [comment]
Would it be useful for the gnucobol3 repo to re-import (if not possible otherwise) with enabled committer translation? Just have a look the history nbert is not nbert :-)

**Comment by lefessan** [comment]
It's a bit weird, I didn't do anything different from the previous `gnucobol` repo, committer translation was better done the first time. Ok, I will need to investigate what changed and how it can be fixed...

**Comment by GitMensch** [comment]
Ping @lefessan the current mirroring seem to not translate (any more?)
Neither sf-mensch nor ddeclerck are transformed in the git branches

Note: if you haven't the translation from "chaat" to @chuck-haatvedt you may want to add it.

**Comment by GitMensch** [comment]
@lefessan the repo and branches are fine now, only open point is the commiter-id part (and possibly importing old commits/tags, maybe even including the "old" history - but anything from the single upstream svn would already be useful for git blame).

**Comment by GitMensch** [comment]
I'm closing this as I plan moving upstream to git and codeberg in 2026 - this will possibly have committer translation enabled. Afterwards the gitside branches may be moved there as well or a direct mirror be setup...
